### DISCUSSION TITLE
Context menus don't come up on Android

### DISCUSSION THREAD

**Discussion by AArnott** [AvaloniaUI/Avalonia#10472] [Q&A]
My xaml file describes a ContextMenu and applies it via styling to ListBox items. It works via right-click on desktop (Windows), but on Android (at least on the emulator, which is all I've tried), long tap and right-clock do not work to bring up the context menu.

Is this a bug, or do I need to do something special for context menus on Android?

I'm using avalonia 11.0.999-cibuild0030643-beta.

```xaml
<UserControl xmlns="https://github.com/avaloniaui"
             xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
             xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
             xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
             xmlns:vm="clr-namespace:ClueBuddy.ViewModels;assembly=ClueBuddy"
             xmlns:root="using:ClueBuddyAvl"
             mc:Ignorable="d" d:DesignWidth="300" d:DesignHeight="400"
             x:Class="ClueBuddyAvl.Views.CluesView">
    <Design.DataContext>
        <vm:GameViewModel/>
    </Design.DataContext>
    <Panel Name="WrapperPanel">
        <ListBox Items="{Binding Clues}" Background="Transparent">
            <ListBox.Resources>
                <ContextMenu x:Key="ClueContextMenu">
                    <MenuItem Header="✏️ Edit" Command="{Binding DataContext.EditClueCommand, ElementName=WrapperPanel}" CommandParameter="{Binding}" />
                    <MenuItem Header="❌ Delete" Command="{Binding DataContext.DeleteClueCommand, ElementName=WrapperPanel}" CommandParameter="{Binding}" />
                </ContextMenu>
            </ListBox.Resources>
            <ListBox.Styles>
                <Style Selector="Grid.ClueItem">
                    <Setter Property="ContextMenu" Value="{StaticResource ClueContextMenu}" />
                </Style>
            </ListBox.Styles>
            <ListBox.ItemTemplate>
                <root:ClueViewModelTemplateSelector>
                    <root:ClueViewModelTemplateSelector.Hand>
                        <DataTemplate>
                            <Grid Classes="ClueItem">
                                <Grid.ColumnDefinitions>
                                    <ColumnDefinition Width="auto" />
                                    <ColumnDefinition Width="5" />
                                    <ColumnDefinition Width="*" />
                                    <ColumnDefinition Width="auto" />
                                </Grid.ColumnDefinitions>
                                <TextBlock Text="✋" />
                                <StackPanel Grid.Column="2">
                                    <StackPanel Orientation="Horizontal">
                                        <TextBlock Text="{Binding Clue.Player.Name}" />
                                        <TextBlock Text=" holds " />
                                    </StackPanel>
                                    <ItemsControl Items="{Binding Clue.Cards}" Margin="15,0,0,0">
                                        <ItemsControl.ItemTemplate>
                                            <DataTemplate>
                                                <TextBlock TextWrapping="Wrap" Text="{Binding Name}" />
                                            </DataTemplate>
                                        </ItemsControl.ItemTemplate>
                                    </ItemsControl>
                                </StackPanel>
                                <ContentControl Classes="Conflict" Grid.Column="3" />
                            </Grid>
                        </DataTemplate>
                    </root:ClueViewModelTemplateSelector.Hand>
                </root:ClueViewModelTemplateSelector>
            </ListBox.ItemTemplate>
        </ListBox>
    </Panel>
</UserControl>
```

**Comment by maxkatz6** [comment]
New Holding event was implemented in 11.0.0-preview5 and it should raise ContextRequested event (which opens ContextFlyout or in your case ContextMenu).
https://github.com/AvaloniaUI/Avalonia/pull/9652/files#diff-46f1b9186b3c21f2f39416bc504985f8766f2b4d5c35cc284b546e9916179259R378

cc @emmauss

**Reply by AArnott** [reply]
So this morning I tested it on a physical device, and it worked, though apparently inconsistently. Maybe it's an issue of exactly where I long tap. I'd like anywhere in the rectangular grid to work as a long tap target, but I wonder if it's only on (or off?) the text blocks.
Then I went back to the emulator, and it apparently works there too. Go figure.

**Reply by AArnott** [reply]
Although on the emulator, the rendering is odd, in that the emojis I'm using have inconsistent widths (Windows didn't have this issue) and even more odd, it shows a scroll button as if there was more, but there isn't.

![image](https://user-images.githubusercontent.com/3548/221366212-da3764f4-121c-4ab4-a23b-4a53cf19864a.png)

**Comment by emmauss** [comment]
> My xaml file describes a ContextMenu and applies it via styling to ListBox items. It works via right-click on desktop (Windows), but on Android (at least on the emulator, which is all I've tried), long tap and right-clock do not work to bring up the context menu.
> 
> Is this a bug, or do I need to do something special for context menus on Android?
> 
> I'm using avalonia 11.0.999-cibuild0030643-beta.

May I see how you are defining the context menu

**Reply by AArnott** [reply]
I've added the xaml to the original post.

**Reply by timunie** [reply]
I think emojis are different on each device, if the font isn't explicitly set. i noticed the same on web github page and my github App on mobile. Same for twitter and signal-App.
